src/core_modules/capture-core/metaDataMemoryStoreBuilders/common/factory/searchGroup/SearchGroupFactory.js

```diff
@@ -81,35 +90,52 @@ class SearchGroupFactory {
     }
 
     async _buildElement(searchAttribute: SearchAttribute) {
-        const trackedEntityAttribute = searchAttribute.trackedEntityAttribute;
         const element = new DataElement((o) => {
-            o.id = trackedEntityAttribute.id;
-            o.name = this._getAttributeTranslation(
-                trackedEntityAttribute.translations, translationPropertyNames.NAME) ||
-                trackedEntityAttribute.displayName;
-            o.shortName = this._getAttributeTranslation(
-                trackedEntityAttribute.translations, translationPropertyNames.SHORT_NAME) ||
-                trackedEntityAttribute.displayShortName;
-            o.formName = this._getAttributeTranslation(
-                trackedEntityAttribute.translations, translationPropertyNames.NAME) ||
-                trackedEntityAttribute.displayName;
-            o.description = this._getAttributeTranslation(
-                trackedEntityAttribute.translations, translationPropertyNames.DESCRIPTION) ||
-                trackedEntityAttribute.description;
+            const {
+                id,
+                translations,
+                displayName,
+                displayShortName,
+                description,
+                unique,
+                valueType,
+            } = searchAttribute.trackedEntityAttribute;
+
+            o.id = id;
+            o.name =
```

Comment:
thats of course a personal preference. If you feel that the previous version has some advantages (readability, code quality etc) please let me know and I am happy to discuss/revert the changes. :)

Reply:
I think both are fine. Pretty sure it's more efficient to not do the desctructering in most engines.